capplets/appearance/appearance-font.c

```diff
@@ -781,6 +802,40 @@ void font_init(AppearanceData* data)
 			  G_CALLBACK (marco_changed),
 			  data);
 
+	/*In a wayland session we must manage MATE font settings for xwayland
+	 *and also manage GNOME font settings for native wayland applications
+	 *
+	 *First find out if we are running under a wayland session
+	 */
+
+	ret = system ("killall -0 -e Xwayland");
+	/*If we are, set the GNOME interface keys too*/
+	if (ret == 0)
+	{
+		widget = appearance_capplet_get_widget(data, "application_font");
+		g_settings_bind (data->interface_gnome_settings,
+				 GTK_FONT_KEY,
+				 G_OBJECT (widget),
+				 "font-name",
+				 G_SETTINGS_BIND_DEFAULT);
```

Comment:
how does this interact if one of the settings change (either the GNOME or MATE one) from outside?  does it sync both, or does this goes crazy?  I would think it would just sync both but better safe than sorry

Reply:
BTW if it works, maybe we could just bind the two regardless of Wayland?  Does it cause any issue on X11?  It's just that MCC would partially control the GNOME settings as well, but given some apps use those maybe it's only fair?

Reply:
(BTW IMO we should at some point have MSD work under Wayland, but I get that it's very far from trivial, so maybe it's not realistic, at least not for now…)

Reply:
All the code does here is write to gsettings. I just checked, having this and dconf-editor open at the same time. If you change it in either one it is applied and immediately reflected in the other.

Binding the GNOME settings when under x11 would probably get complaints from users and even distros, as no doubt some want to have both sessions installed and with different settings. This is no problem under x11 but is impossible under wayland at this time unless we maintain our own compositor (probably a fork of wayfire but with really very few changes). Probably we should submit a PR for wayfire to allow use of MATE or other gsettings for these values

Reply:
Also note that we DO in fact need to run mate-settings-daemon in the x11 session or we don't apply nondefault theme/font/icon theme values in Xwayland apps. It controls only settings of apps running under x11 or xwayland, but without it we can no longer sync the theme etc between a wayland session and apps running under xwayland. Therefore, we run m-s-d under the current wayland session for that purpose

Reply:
> Probably we should submit a PR for wayfire to allow use of MATE or other gsettings for these values.

Yeah I don't know how wayfire works yet, but possibly a plugin similar to Compiz's MATE integration?  Anyway yes if it's at this level that it has to be modified it'd make sense to do so there.  But I guess that can be delayed if you prefer.